### PR TITLE
Feat/lesson02

### PR DESCRIPTION
**Pull request by sinkevichdenis**

### CODE REVIEW

File: lesson_02/routes/app.route.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+    try {
+        res.status(201).json({ message: 'User list loaded', users: Object.values(store) });
+    } catch (e) {
+        handleError(res, e);
+    }
+});
+
+router.post('/', async (req, res) =>  {
+    try {
+        const { id, login } = req.body;
```
Comment (Ivanko5417) [inline lesson_02/routes/app.route.js:19]: id generation must be on the server side

File: lesson_02/routes/app.route.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import { Router } from 'express';
+import { addUserSchema, updateUserSchema, removeUserSchema } from '../schemas/user.schema';
+import { getLoginMatch, handleError } from '../utils';
+
+export const router = Router();
+
+const store = {};
+
+router.get('/list', (req, res) =>  {
```
Comment (Ivanko5417) [inline lesson_02/routes/app.route.js:9]: according to the task here should be getting user by id endpoint

File: lesson_02/routes/app.route.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
+import { Router } from 'express';
+import { addUserSchema, updateUserSchema, removeUserSchema } from '../schemas/user.schema';
+import { getLoginMatch, handleError } from '../utils';
+
+export const router = Router();
+
+const store = {};
```
Comment (Ivanko5417) [inline lesson_02/routes/app.route.js:7]: it's not necessary to fix it now, but it's better to have a separate module which will have logic related to store

File: nodemon.json [hunk 4]
```diff
@@ -0,0 +1,6 @@
+{
+  "watch": ["src"],
+  "ext": ".ts,.js",
+  "ignore": [],
+  "exec": "ts-node"
```
Comment (Ivanko5417) [inline nodemon.json:5]: let's add ts-node to the `package.json`

File: lesson_02/utils.js [hunk 5]
```diff
@@ -0,0 +1,10 @@
+export const handleError = (res, e) => {
+    if (e.isJoi) {
+        res.status(422).json({ message: e.message });
```
Comment (Ivanko5417) [inline lesson_02/utils.js:3]: in case of joi error, let's return 400 status code

File: lesson_02/routes/app.route.js (hunk 2 above)
Comment (Ivanko5417) [reply lesson_02/routes/app.route.js:9]: also, you didn't add `getAutoSuggestUsers` endpoint

File: lesson_02/schemas/user.schema.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    id: Joi.string().required(),
+    login: Joi.string().required(),
+    password: Joi.string().pattern(new RegExp('[a-zA-Z0-9]')).required(),
+    age: Joi.number().integer().min(4).max(130).required(),
+    isDeleted: Joi.boolean().required()
+});
+
+export const updateUserSchema = Joi.object({
+    id: Joi.any(),
+    login: Joi.string().required(),
```
Comment (Ivanko5417) [inline lesson_02/schemas/user.schema.js:13]: let's allow only numbers and letters for login

File: lesson_02/app.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+app.use(express.json({ extended: true }));
+app.use('/', router);
+
+app.listen(PORT, () => {
+    console.log(`App is listening on port http://localhost:${PORT}`);
+});
+
+
+/*
```
Comment (Ivanko5417) [inline lesson_02/app.js:16]: let's remove unnecessary code

**Review by Ivanko5417 — CHANGES_REQUESTED** [review]

File: lesson_02/routes/app.route.js (hunk 1 above)
Comment (sinkevichdenis) [reply lesson_02/routes/app.route.js]: Done

File: lesson_02/routes/app.route.js (hunk 2 above)
Comment (sinkevichdenis) [reply lesson_02/routes/app.route.js:7]: Done

File: lesson_02/routes/app.route.js (hunk 3 above)
Comment (sinkevichdenis) [reply lesson_02/routes/app.route.js]: Done

File: lesson_02/utils.js (hunk 5 above)
Comment (sinkevichdenis) [reply lesson_02/utils.js]: Done

File: lesson_02/schemas/user.schema.js (hunk 6 above)
Comment (sinkevichdenis) [reply lesson_02/schemas/user.schema.js]: Done

File: lesson_02/app.js (hunk 7 above)
Comment (sinkevichdenis) [reply lesson_02/app.js]: Done

File: nodemon.json (hunk 4 above)
Comment (sinkevichdenis) [reply nodemon.json:5]: Could you clarify this moment?

**Review by Ivanko5417 — APPROVED** [review]

**Pull request merged**